services/sqs/src/main/java/software/amazon/awssdk/services/sqs/batchmanager/SqsBatchManager.java

```diff
@@ -134,7 +134,7 @@ static Builder builder() {
          * @param executor the executor to be used.
          * @returna reference to this object so that method calls can be chained together.
          */
-        Builder executor(Executor executor);
+        Builder executor(ExecutorService executor);
```

Comment:
Hmm, why did we revert it back to `Executor`?  If lack of shutdown is the reason, we can cast the created executor to executor service and call shutdown. https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/AttributeMap.java#L91-L93

Reply:
I originally changed it to `Executor` when I was trying to access the client's `FUTURE_COMPLETION_EXECUTOR`. Since they sync client doesn't have that executor, it was ok to revert back to an `ExecutorService`.